src/main/java/com/aws/greengrass/security/SecurityService.java

```diff
@@ -172,6 +180,30 @@ public static URI uriFromPossibleFileURIString(String path) {
         }
     }
 
+    /**
+     * Get KeyManagers for the default device identity.
+     *
+     * @return key managers
+     * @throws TLSAuthException if any error happens
+     */
+    @SuppressWarnings({"PMD.AvoidCatchingGenericException", "PMD.PreserveStackTrace"})
+    public KeyManager[] getDeviceIdentityKeyManagers() throws TLSAuthException {
+        URI privateKey = getDeviceIdentityPrivateKeyURI();
+        URI certPath = getDeviceIdentityCertificateURI();
+        try {
+            return RetryUtils.runWithRetry(SECURITY_SERVICE_RETRY_CONFIG, () -> getKeyManagers(privateKey, certPath),
+                    "get-key-managers", logger);
+        } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)